pdata/ptrace/traces.go

```diff
@@ -12,67 +12,49 @@
+package ptrace // import "go.opentelemetry.io/collector/pdata/ptrace"
 
 import (
+	"go.opentelemetry.io/collector/pdata/internal"
 	otlpcollectortrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 )
 
-// TracesToOtlp internal helper to convert Traces to otlp request representation.
-func TracesToOtlp(mw Traces) *otlpcollectortrace.ExportTraceServiceRequest {
-	return mw.orig
-}
-
-// TracesFromOtlp internal helper to convert otlp request representation to Traces.
-func TracesFromOtlp(orig *otlpcollectortrace.ExportTraceServiceRequest) Traces {
-	return Traces{orig: orig}
-}
-
-// TracesToProto internal helper to convert Traces to protobuf representation.
-func TracesToProto(mw Traces) otlptrace.TracesData {
-	return otlptrace.TracesData{
-		ResourceSpans: mw.orig.ResourceSpans,
-	}
-}
+// Traces is the top-level struct that is propagated through the traces pipeline.
+// Use NewTraces to create new instance, zero-initialized instance is not valid for use.
+type Traces internal.Traces
 
-// TracesFromProto internal helper to convert protobuf representation to Traces.
-func TracesFromProto(orig otlptrace.TracesData) Traces {
-	return Traces{orig: &otlpcollectortrace.ExportTraceServiceRequest{
-		ResourceSpans: orig.ResourceSpans,
-	}}
+func newTraces(orig *otlpcollectortrace.ExportTraceServiceRequest) Traces {
+	return Traces(internal.NewTraces(orig))
 }
 
-// Traces is the top-level struct that is propagated through the traces pipeline.
-// Use NewTraces to create new instance, zero-initialized instance is not valid for use.
-type Traces struct {
-	// When marhsal/unmarshal unless it is in the request for otlp protocol, convert to otlptrace.TracesData.
-	orig *otlpcollectortrace.ExportTraceServiceRequest
+func (ms Traces) getOrig() *otlpcollectortrace.ExportTraceServiceRequest {
+	return internal.GetOrigTraces(internal.Traces(ms))
 }
 
 // NewTraces creates a new Traces struct.
 func NewTraces() Traces {
-	return Traces{orig: &otlpcollectortrace.ExportTraceServiceRequest{}}
+	return newTraces(&otlpcollectortrace.ExportTraceServiceRequest{})
 }
 
 // MoveTo moves all properties from the current struct to dest
 // resetting the current instance to its zero value.
-func (td Traces) MoveTo(dest Traces) {
-	*dest.orig = *td.orig
-	*td.orig = otlpcollectortrace.ExportTraceServiceRequest{}
+func (ms Traces) MoveTo(dest Traces) {
+	*dest.getOrig() = *ms.getOrig()
+	*ms.getOrig() = otlpcollectortrace.ExportTraceServiceRequest{}
 }
 
 // Clone returns a copy of Traces.
-func (td Traces) Clone() Traces {
+func (ms Traces) Clone() Traces {
 	cloneTd := NewTraces()
-	td.ResourceSpans().CopyTo(cloneTd.ResourceSpans())
+	ms.ResourceSpans().CopyTo(cloneTd.ResourceSpans())
 	return cloneTd
 }
 
 // SpanCount calculates the total number of spans.
-func (td Traces) SpanCount() int {
+func (ms Traces) SpanCount() int {
```

Comment:
Why receiver var name is changed to `ms`?

Reply:
Because all generated things are "ms", so was easier to copy paste helper funcs :))